### PR TITLE
[WIP] Add support for ChatGPT conversations via Matrix threads

### PR DESCRIPTION
**Pull request by bertybuttface**
Please don't merge until WIP tags are removed. 

Closes https://github.com/jakecoppinger/matrix-chatgpt-bot/issues/5

### CODE REVIEW

File: src/handlers.ts [hunk 1]
```diff
@@ -30,14 +54,42 @@ Please add me to an unencrypted chat.`);
...
+      }
+
+      let storageKey = event.event_id // store event_id of the received message, it will be the message root if we just started a new thread.
+      console.log('Attempting to save event_id')
+      if (event.content["m.relates_to"] !== undefined) {
+        console.log('Detected existing relates_to saving root_event_id instead.')
+        storageKey = event.content["m.relates_to"].event_id // it is possible this thread already existed so we might need an older root, check for one.
+      }
+  
+      let result
```
Comment (jakecoppinger) [inline src/handlers.ts:77]: nit: could we add types to variable initialisations?

**Comment by bertybuttface** [comment]
Yeah this is first draft status, it works but it is messy.

File: src/handlers.ts (hunk 1 above)
Comment (bertybuttface) [reply src/handlers.ts]: Fixed.

**Comment by bertybuttface** [comment]
The remaining thing to test is a client without modern Threads support.

**Pull request closed without merging**